From 4f649aaeaf3a8300bd9fd1130007b12770593d46 Mon Sep 17 00:00:00 2001
From: Nikola Pajkovsky <npajkovs@redhat.com>
Date: Fri, 22 Jul 2011 17:30:09 +0200
Subject: [PATCH 26/26] rhbz#724999 - fix (null) in summary

when you report selinux problem via libreport summary looks like

[abrt] (null): SELinux is preventing /usr/bin/abrt-dump-oops from 'open'
accesses on the file /var/log/messages.

(null) is name of the package, but it doesn't have any. Reasonable summary
is only from 'reason' file.

Signed-off-by: Nikola Pajkovsky <npajkovs@redhat.com>
---
 src/plugins/rhbz.c |   27 ++++++++++++++++-----------
 1 files changed, 16 insertions(+), 11 deletions(-)

diff --git a/src/plugins/rhbz.c b/src/plugins/rhbz.c
index 0a4c28e..5b53465 100644
--- a/src/plugins/rhbz.c
+++ b/src/plugins/rhbz.c
@@ -307,20 +307,25 @@ int rhbz_new_bug(struct abrt_xmlrpc *ax, problem_data_t *problem_data,
                                                                 FILENAME_TAINTED_SHORT);
 
     struct strbuf *buf_summary = strbuf_new();
-    strbuf_append_strf(buf_summary, "[abrt] %s", package);
-
-    if (function != NULL && strlen(function) < 30)
-        strbuf_append_strf(buf_summary, ": %s", function);
-
-    if (reason != NULL)
-        strbuf_append_strf(buf_summary, ": %s", reason);
-
-    if (tainted_short && analyzer
-        && (strcmp(analyzer, "Kerneloops") == 0))
+    if (analyzer && strcmp(analyzer, "libreport") == 0)
     {
-        strbuf_append_strf(buf_summary, ": TAINTED %s", tainted_short);
+        strbuf_append_str(buf_summary, reason);
     }
+    else
+    {
+        strbuf_append_strf(buf_summary, "[abrt] %s", package);
+        if (function && strlen(function) < 30)
+            strbuf_append_strf(buf_summary, ": %s", function);
 
+        if (reason)
+            strbuf_append_strf(buf_summary, ": %s", reason);
+
+        if (tainted_short && analyzer
+            && (strcmp(analyzer, "Kerneloops") == 0))
+        {
+            strbuf_append_strf(buf_summary, ": TAINTED %s", tainted_short);
+        }
+    }
     char *status_whiteboard = xasprintf("abrt_hash:%s", duphash);
 
     char *bz_dsc = make_description_bz(problem_data);
-- 
1.7.6

